cdap-ui/app/cdap/components/AbstractWidget/InputKeyValueDropdown/InputKeyValueDropdown.tsx

```diff
@@ -0,0 +1,181 @@
+  value: string | number;
+  label: string;
+}
+
+export type IDropdownOption = string | number | IComplexDropdown;
+
+interface IInputKeyValueDropdownProps extends IAbstractRowProps<typeof styles> {
+  keyPlaceholder?: string;
+  kvDelimiter?: string;
+  // dropdownOptions: IDropdownOption[];
+  dropdownOptions: any;
+  extraConfig?: any;
+}
+
+interface IKeyValueState {
+  value: string;
+  key: string;
+}
+
+type StateKeys = keyof IKeyValueState;
+
+class InputKeyValueDropdown extends AbstractRow<IInputKeyValueDropdownProps, IKeyValueState> {
+  public static defaultProps = {
+    keyPlaceholder: 'Test',
+    kvDelimiter: '=',
+    dropdownOptions: [],
+    delimiter: ',',
+  };
+
+  public state = {
+    key: '',
+    value: '',
+  };
+
+  public componentDidMount() {
+    const [key = '', value = ''] = this.props.value.split(this.props.kvDelimiter);
+
+    this.setState({
+      key,
+      value,
+    });
+  }
+
+  private handleChange = (type: StateKeys, e) => {
+    this.setState(
+      {
+        [type]: e.target.value,
+      } as Pick<IKeyValueState, StateKeys>,
+      () => {
+        const key = this.state.key;
+        const value = this.state.value;
+
+        const updatedValue = key.length > 0 ? [key, value].join(this.props.kvDelimiter) : '';
+        this.onChange(updatedValue);
+      }
+    );
+  };
+
+  public getFields(schemas: IStageSchema[], allowedTypes: string[]) {
+    let fields = [];
+    if (!schemas || schemas.length === 0) {
+      return fields;
+    }
+    const stage = schemas[0];
+
+    try {
+      const unparsedFields = JSON.parse(stage.schema).fields;
+
+      if (unparsedFields.length > 0) {
+        fields = unparsedFields
+          .filter((field: IField) => this.containsType(field.type, allowedTypes))
+          .map((field: IField) => field.name);
+      }
+    } catch {
+      // tslint:disable-next-line: no-console
+      console.log('Error: Invalid JSON schema');
+    }
+    return fields;
+  }
+
+  // Function that checks if types contains a type that is in allowedTypes
+  // This is meant to handle nullable fields since a nullable string type is
+  // presented as ['string','null'].
+  public containsType(types: string[], allowedTypes: string[]) {
+    if (allowedTypes.length === 0) {
+      return true;
+    }
+
+    return allowedTypes.includes(this.extractType(types));
+  }
+
+  public extractType(types) {
+    let value = types;
+    if (types instanceof Array) {
+      if (types.length === 1) {
+        value = types[0];
+      } else if (types.length === 2 && types.includes('null')) {
+        value = types.indexOf('null') === 0 ? types[1] : types[0];
+      }
+    }
+
+    if (typeof value === 'object') {
+      value = value.logicalType || value;
+    }
+    return value;
+  }
+
+  public renderInput = () => {
+    // const inputSchema = objectQuery(this.props.extraConfig, 'inputSchema');
+    const fieldValues = this.getFields(this.props.dropdownOptions, []);
```

Comment:
This doesn't look right. The `dropdownOptions` comes from `widgetProps` which means its a static one from the widget json (like placeholder or default value). 

Based on my understanding shouldn't this be list of fields from the input schema? If so then the commented out code seems right as this widget needs to show fields from current input schema. Did I understand that right?